internal/xenia/pipeline.go

```diff
@@ -9,13 +9,16 @@ import (
 	"github.com/ardanlabs/kit/db"
 	"github.com/ardanlabs/kit/db/mongo"
 	"github.com/ardanlabs/kit/log"
+	"github.com/cayleygraph/cayley"
+	"github.com/coralproject/shelf/internal/wire"
 	"github.com/coralproject/shelf/internal/xenia/query"
+	"github.com/pborman/uuid"
 	"gopkg.in/mgo.v2"
 	"gopkg.in/mgo.v2/bson"
 )
 
 // execPipeline executes the sepcified pipeline query.
-func execPipeline(context interface{}, db *db.DB, q *query.Query, vars map[string]string, data map[string]interface{}, explain bool) (docs, []map[string]interface{}, error) {
+func execPipeline(context interface{}, db *db.DB, graph *cayley.Handle, q *query.Query, vars map[string]string, data map[string]interface{}, explain bool) (docs, []map[string]interface{}, error) {
```

Comment:
The problem is that we don't have access to the DB config anywhere in here and we can't get the hostname etc. from *db.DB.  I originally tried that, but that's not how the kit is set up.  We could change the kit, to be able to retrieve host information, etc. from the *db.DB value.  Otherwise, we would have to pass this in.